modules/SerialCommHub/tiny_modbus_rtu.hpp

```diff
@@ -57,18 +55,27 @@ class TinyModbusRTU {
     ~TinyModbusRTU();
 
     bool open_device(const std::string& device, int baud, bool ignore_echo,
-                     const Everest::GpioSettings& rxtx_gpio_settings, const Parity parity);
+                     const Everest::GpioSettings& rxtx_gpio_settings, const Parity parity,
+                     std::chrono::milliseconds initial_timeout, std::chrono::milliseconds within_message_timeout);
+
     std::vector<uint16_t> txrx(uint8_t device_address, FunctionCode function, uint16_t first_register_address,
-                               uint16_t register_quantity, bool wait_for_reply = true,
+                               uint16_t register_quantity, uint16_t chunk_size, bool wait_for_reply = true,
                                std::vector<uint16_t> request = std::vector<uint16_t>());
 
 private:
     // Serial interface
     int fd{0};
     bool ignore_echo{false};
+
+    std::vector<uint16_t> txrx_impl(uint8_t device_address, FunctionCode function, uint16_t first_register_address,
+                                    uint16_t register_quantity, bool wait_for_reply = true,
+                                    std::vector<uint16_t> request = std::vector<uint16_t>());
+
     int read_reply(uint8_t* rxbuf, int rxbuf_len);
 
     Everest::Gpio rxtx_gpio;
+    std::chrono::milliseconds initial_timeout;
+    std::chrono::milliseconds within_message_timeout;
```

Comment:
Both could be const and initialized in the initializer list

Reply:
This would be a better design. But in the way the tiny_modbus implemented and used it would require quite a refactoring: 
* there is no constructor defined for TinyModbusRTU
* the object `modbus` is default-constructed in SerialComHub, and the device is opened later on.
So we will need to figure out when and with which parameters to call the TinyModbusRTU constructor

Do you think it makes sense to add this refactoring into this MR?